platform/retarget.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * mbed Microcontroller Library
+ * Copyright (c) 2006-2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * This file contains symbols used by the implementation that are not present
+ * in the armcc errno.h, or sys/stat.h. These are used in for the POSIX
+ * filesystem API to return errno codes, for example.
+ */
+
+#ifndef RETARGET_H
+#define RETARGET_H
+
+#if defined TOOLCHAIN_ARM_STD || defined TOOLCHAIN_IAR
```

Comment:
Yes, the intent is to unify errno values across toolchains, and supply values where missing. I've updated comments and changed to use compiler emitted macros.